Update to Bevy 0.12.

pcwalton opened the pull request:
Everything seems to work in both the simple example and in my own app.

The one thing I wasn't sure about was whether to rename the `queue_` methods to `prepare_` now that they've been moved to the `PrepareBindGroups` render set. They had to be moved there because in Bevy 0.12 `Queue` executes before `Prepare`, whereas this crate expected them to execute in the other order.

IceSentry commented on src/render/mod.rs:
```diff
@@ -724,7 +723,8 @@ pub fn render_app_builder(app: &mut App) {
             Render,
-            (queue_infinite_grids, queue_grid_view_bind_groups).in_set(RenderSet::Queue),
+            (queue_infinite_grids, queue_grid_view_bind_groups)
+                .in_set(RenderSet::PrepareBindGroups),
```
I'm not sure I understand why this is necessary. I understand that the sets were reordered but the Queue set should still be used to queue things. I haven't touched this code that much though so it's possible it's doing something incompatible with that.

pcwalton replied:
The problem was that you put bind group creation and queuing into the same system, while Bevy 0.12 wants them to be in separate systems. I've split up the logic appropriately.

IceSentry replied:
Ah, thank you for the investigation, I didn't implement the crate originally so I wasn't aware of that.

IceSentry reviewed (commented):
If the queue -> prepare set change is required then I'd prefer if the systems were renamed.

Other than that LGTM. I'd prefer if the bind group entries api was used, but I'll do that once this PR is merged.

pcwalton commented:
OK, I discovered that the shadow was broken, so this PR fixes that too. That required some fairly large surgery. I've also updated the PR to use the `bind_group_entries` API. Finally, I renamed the `queue` methods to `prepare` and/or split them out into separate methods as appropriate.

This should be ready to go now.

IceSentry reviewed (approved):
Thank you very much for all the work!

aevyrie commented:
Thanks @pcwalton!
